replica/compaction_group.hh

```diff
@@ -135,6 +135,14 @@ public:
     }
 };
 
+class compaction_group_manager {
+public:
+    virtual ~compaction_group_manager() {}
+    virtual std::vector<std::unique_ptr<compaction_group>> make_compaction_groups() const = 0;
```

Comment:
We should switch to chunked_vector<> to avoid large allocation, because we aim for 100k tablets.